Task3_1/Queue.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Task3_1
+{
+    internal class Queue<T> : IQueue<T>
+    {
+        private int _size;
+        private int _maximumCapacity;
+        private T[] _queue;
+
+        public Queue()
+        {
+            _size = 0;
```

Comment:
_queue = new T[_maximumCapacit], allocate memory once and reuse it (see next comment)

Reply:
Modified